File: By-Python/leetcode/jump-game.py
def canJump(self, nums: List[int]) -> bool:
    lp, rp = 0, 0
    destination = len(nums)-1
    
    while rp < destination:
        if lp > rp:
            return False
        
        furthest = max([i+nums[i] for i in range(lp, rp+1)])
        lp, rp = rp+1, furthest
        
    return True

File: By-Swift/By-Swift/Leetcode/jump-game.swift
class Solution {
    func canJump(_ nums: [Int]) -> Bool {
        var pointer = (0,0)
        var furthest = 0
        let destination = nums.count-1
        
        while pointer.1 < destination {
            if pointer.0 > pointer.1 {
                return false
            }
            
            for i in pointer.0...pointer.1 {
                furthest = max(furthest, i+nums[i])
            }
            
            pointer = (pointer.1+1, furthest)
        }
        return true
    }
}   